src/CTA.WebForms2Blazor/ClassConverters/GlobalClassConverter.cs

```diff
@@ -1,41 +1,197 @@
+
+                    _configureMethodStatements = _configureMethodStatements.Concat(newStatements);
+                }
+                else if (method.IsEventHandler(ApplicationEndMethodName) || method.IsEventHandler(SessionStartMethodName) || method.IsEventHandler(SessionEndMethodName))
+                {
+                    CommentOutMethod(method);
+                }
+                else
+                {
+                    _keepableMethods = _keepableMethods.Append(method);
+                }
+            }
+
+            // We added all discovered middleware methods as lambdas so global has been
+            // processed as a middleware source
+            _lifecycleManager.NotifyMiddlewareSourceProcessed();
+        }
+
+        private async Task<IEnumerable<UsingDirectiveSyntax>> GetAllUsingStatements()
+        {
+            var typeRequiredNamespaceNames = _sourceFileSemanticModel
+                .GetNamespacesReferencedByType(_originalDeclarationSyntax)
+                .Select(namespaceSymbol => namespaceSymbol.ToDisplayString());
+
+            // Merging required using statements for middleware and source type contents
+            try
+            {
+                var middlewareNamespaceNames = await _taskManager.ManagedRun(_taskId, (token) => _lifecycleManager.GetMiddlewareNamespaces(token));
+                return CodeSyntaxHelper.BuildUsingStatements(typeRequiredNamespaceNames.Union(middlewareNamespaceNames));
+            }
+            catch (OperationCanceledException e)
+            {
+                // TODO: Log this failure
+
+                return CodeSyntaxHelper.BuildUsingStatements(typeRequiredNamespaceNames)
+                    .AddComment(string.Format(Constants.OperationFailedCommentTemplate, AddMiddlewareUsingsOperation));
+            }
+        }
+
+        private void HandleLifecycleMethod(MethodDeclarationSyntax methodDeclaration, WebFormsAppLifecycleEvent lifecycleEvent)
+        {
+            var statements = methodDeclaration.Body.Statements;
+            var lambdaExpression = LifecycleManagerService.ContentIsPreHandle(lifecycleEvent) ?
+                MiddlewareSyntaxHelper.BuildMiddlewareLambda(preHandleStatements: statements) :
+                MiddlewareSyntaxHelper.BuildMiddlewareLambda(postHandleStatements: statements);
+
+            _lifecycleManager.RegisterMiddlewareLambda(lifecycleEvent, lambdaExpression);
+        }
+
+        private void CommentOutMethod(MethodDeclarationSyntax methodDeclaration)
+        {
+            // Get the method as a series of strings and give the reason it was removed
+            var newComments = methodDeclaration.AsStringsByLine().Prepend(Constants.UnusableCodeComment);
+
+            if (_endOfClassComments.Count() > 0)
+            {
+                // Prepend an extra space to get separation from last commented method
+                newComments = newComments.Prepend(string.Empty);
+            }
```

Comment:
nit: `.Count() > 0` can be replaced with `.Any()`. These are functionally equivalent, but the latter is more expressive.

Reply:
Prepending an empty string actually does work because the AddComments extension method treats every string as a specific line when a collection is passed, so in this case the "" will render as an extra line with just "//" between a series of commented out methods

Reply:
Understood, appreciate the clarification